graph/src/data/store/mod.rs

```diff
@@ -647,6 +649,38 @@ pub trait TryIntoEntityIterator<E>: IntoIterator<Item = Result<(Word, Value), E>
 
 impl<E, T: IntoIterator<Item = Result<(Word, Value), E>>> TryIntoEntityIterator<E> for T {}
 
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum EntityValidationError {
+    #[error("The provided entity has fields not defined in the schema for entity `{0}`: {1}")]
+    FieldsNotDefined(String, String),
+
+    #[error("Entity {0}[{1}]: unknown entity type `{0}`")]
+    UnknownEntityType(String, String),
+
+    #[error("Entity {0}[{1}]: field `{2}` is of type {3}, but the value `{4}` contains a {5} at index {6}")]
+    MismatchedElementTypeInList(String, String, String, String, String, String, usize),
+
+    #[error(
+        "Entity {0}[{1}]: the value `{2}` for field `{3}` must have type {4} but has type {5}"
+    )]
+    InvalidFieldType(String, String, String, String, String, String),
```

Comment:
especially this one xD

Reply:
Thanks for catching those, I was blindly following how error struct from other parts of the [code](https://github.com/graphprotocol/graph-node/blob/48e0958e505b95a0d59ee6b3ef3d874cacda97eb/graph/src/schema/mod.rs#L69).

But this one clearly needs to be named. Thanks will keep an eye for such things going forward